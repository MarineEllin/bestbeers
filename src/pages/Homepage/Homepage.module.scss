@use "assets/styles/mixins" as mixin;
.homepageContainer {
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  .homepageContent {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    .grid {
      display: grid;
      place-content: center center;
      grid-template-columns: repeat(4, 350px);
      gap: 20px;
      @include mixin.xs-medium {
        grid-template-columns: 1fr;
      }
      @include mixin.xs-medium {
        grid-template-columns: 1fr;
      }
      @include mixin.large {
        grid-template-columns: repeat(2, 350px);
      }
      @include mixin.xl {
        grid-template-columns: repeat(3, 350px);
      }
      @include mixin.xxl {
        grid-template-columns: repeat(4, 350px);
      }
    }
    .loadMoreBeers {
      margin: 20px;
      padding: 20px;
      background-color: var(--primary-light);
      border: 1px solid var(--primary-light);
      color: var(--dark);
      border-radius: var(--border-radius);
      transition: 0.5s;
      cursor: pointer;
      &:hover {
        background-color: var(--light);
      }
    }
    .loadMoreBeers {
      margin: 20px;
      padding: 20px;
      background-color: var(--primary-light);
      border: 1px solid var(--primary-light);
      color: var(--dark);
      border-radius: var(--border-radius);
      transition: 0.5s;
      cursor: pointer;
      &:hover {
        background-color: var(--light);
      }
    }
  }
}
