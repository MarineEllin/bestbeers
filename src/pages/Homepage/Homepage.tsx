import { BeerObject } from "models/beer.model";
import { useRecoilState, useRecoilValue } from "recoil";
import { beersListState, pageState } from "recoilStates/atom";
import Beer from "pages/Components/Beer/Beer";
import styles from "./Homepage.module.scss";
import { useGetBeers } from "hooks/useGetBeers";
import FilterMenu from "pages/Components/FilterMenu/FilterMenu";

function Homepage() {
  const beersList = useRecoilValue(beersListState);
  const [page, setPage] = useRecoilState(pageState);
  useGetBeers();

  const loadMoreBeers = () => {
    setPage(page + 1);
  };

  return (
    <div className={styles.homepageContainer}>
      <div className={styles.homepageContent}>
        <FilterMenu />
        <div className={styles.grid}>
          {beersList.map((beer: BeerObject) => (
            <Beer key={beer.id} beer={beer} />
          ))}
        </div>
        <div onClick={loadMoreBeers} className={styles.loadMoreBeers}>
          Load more
        </div>
      </div>
    </div>
  );
}

export default Homepage;
