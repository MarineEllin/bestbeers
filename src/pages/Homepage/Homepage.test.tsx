import renderer from "react-test-renderer";
import Homepage from "./Homepage";
import { RecoilRoot } from "recoil";

test("Write a test for Header with snapshot", () => {
  const component = renderer.create(
    <RecoilRoot>
      <Homepage />
    </RecoilRoot>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
