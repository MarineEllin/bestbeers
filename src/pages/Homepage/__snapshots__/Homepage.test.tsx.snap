// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Write a test for Header with snapshot 1`] = `
<div
  className="homepageContainer"
>
  <div
    className="homepageContent"
  >
    <div
      className="menuContainer"
    >
      <i
        className="filterIcon fa-solid fa-filter"
      />
      <div
        className="alcoholContainer"
      >
        <span
          className="mr-5"
        >
          Alcohol by volume 
        </span>
        <button
          aria-pressed={false}
          className="k-button k-button-md k-button-solid k-button-solid-base k-rounded-md"
          onClick={[Function]}
        >
          <span
            className="k-button-text"
          >
            &gt;
             8%
          </span>
        </button>
      </div>
      <div
        className="bitternessContainer"
      >
        <span>
          Bitterness 
        </span>
        <div
          className="k-slider k-slider-horizontal"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          style={
            {
              "gap": 0,
            }
          }
        >
          <div
            className="k-slider-track-wrap"
            style={
              {
                "flexGrow": 1,
                "marginLeft": "0.5rem",
                "marginRight": "0.5rem",
                "position": "relative",
                "touchAction": "none",
              }
            }
          >
            <ul
              className="k-reset k-slider-items"
              style={
                {
                  "margin": 0,
                }
              }
            >
              <li
                className="k-tick k-tick-large k-tick-horizontal"
                style={
                  {
                    "left": "0%",
                    "position": "absolute",
                    "zIndex": 1,
                  }
                }
              >
                <span
                  className="k-label"
                  data-slider-label={true}
                >
                  0
                </span>
              </li>
              <li
                className="k-tick k-tick-large k-tick-horizontal"
                style={
                  {
                    "left": "20%",
                    "position": "absolute",
                    "zIndex": 1,
                  }
                }
              >
                <span
                  className="k-label"
                  data-slider-label={true}
                >
                  20
                </span>
              </li>
              <li
                className="k-tick k-tick-large k-tick-horizontal"
                style={
                  {
                    "left": "40%",
                    "position": "absolute",
                    "zIndex": 1,
                  }
                }
              >
                <span
                  className="k-label"
                  data-slider-label={true}
                >
                  40
                </span>
              </li>
              <li
                className="k-tick k-tick-large k-tick-horizontal"
                style={
                  {
                    "left": "60%",
                    "position": "absolute",
                    "zIndex": 1,
                  }
                }
              >
                <span
                  className="k-label"
                  data-slider-label={true}
                >
                  60
                </span>
              </li>
              <li
                className="k-tick k-tick-large k-tick-horizontal"
                style={
                  {
                    "left": "80%",
                    "position": "absolute",
                    "zIndex": 1,
                  }
                }
              >
                <span
                  className="k-label"
                  data-slider-label={true}
                >
                  80
                </span>
              </li>
              <li
                className="k-tick k-tick-large k-tick-horizontal"
                style={
                  {
                    "left": "100%",
                    "position": "absolute",
                    "zIndex": 1,
                  }
                }
              >
                <span
                  className="k-label"
                  data-slider-label={true}
                >
                  100
                </span>
              </li>
            </ul>
            <div
              className="k-slider-track"
              style={
                {
                  "left": 0,
                  "width": "100%",
                }
              }
            >
              <div
                className="k-slider-selection"
                style={
                  {
                    "width": "100%",
                  }
                }
              />
              <span
                aria-valuemax={100}
                aria-valuemin={0}
                aria-valuenow={100}
                aria-valuetext="100"
                className="k-draghandle"
                role="slider"
                style={
                  {
                    "left": "calc(100%)",
                    "zIndex": 1,
                  }
                }
                tabIndex={0}
                title="Drag"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="grid"
    />
    <div
      className="loadMoreBeers"
      onClick={[Function]}
    >
      Load more
    </div>
  </div>
</div>
`;
